### ISSUE TITLE
unable to generate token for user

### ISSUE THREAD

**Issue by mcertini** [PowerShell/Win32-OpenSSH#1638]
**Troubleshooting steps**
https://github.com/PowerShell/Win32-OpenSSH/wiki/Troubleshooting-Steps

**Terminal issue? please go through wiki**
https://github.com/PowerShell/Win32-OpenSSH/wiki/TTY-PTY-support-in-Windows-OpenSSH

Please answer the following

**"OpenSSH for Windows" version**
7.7.2.2

**Server OperatingSystem**
Windows 10 Pro

**Client OperatingSystem**
Ubuntu 20.04

**What is failing**
Cannot get user to authenticate on SSH server.  Executed FixHostFilePermissions.ps1 and FixUserFilePermissions.ps1 successfully and rebooted the computer.  I am using Putty on Ubuntu.

**Expected output**
I expect to be connected to the server.

**Actual output**
**SSHD Log**
2820 2020-07-22 21:00:07.880 debug2: fd 3 setting O_NONBLOCK
2820 2020-07-22 21:00:07.990 debug3: sock_set_v6only: set socket 3 IPV6_V6ONLY
2820 2020-07-22 21:00:07.990 debug1: Bind to port 22 on ::.
2820 2020-07-22 21:00:07.990 Server listening on :: port 22.
2820 2020-07-22 21:00:07.990 debug2: fd 4 setting O_NONBLOCK
2820 2020-07-22 21:00:07.990 debug1: Bind to port 22 on 0.0.0.0.
2820 2020-07-22 21:00:07.990 Server listening on 0.0.0.0 port 22.
2820 2020-07-22 21:12:17.430 debug3: fd 5 is not O_NONBLOCK
2820 2020-07-22 21:12:17.430 debug3: spawning "C:\\Windows\\System32\\OpenSSH\\sshd.exe" "-R"
2820 2020-07-22 21:12:17.445 debug3: send_rexec_state: entering fd = 8 config len 351
2820 2020-07-22 21:12:17.445 debug3: ssh_msg_send: type 0
2820 2020-07-22 21:12:17.445 debug3: send_rexec_state: done
5848 2020-07-22 21:12:17.695 debug1: inetd sockets after dupping: 3, 3
5848 2020-07-22 21:12:17.695 Connection from 192.168.1.170 port 51461 on 192.168.1.127 port 22
5848 2020-07-22 21:12:17.695 debug1: Client protocol version 2.0; client software version PuTTY_Release_0.73
5848 2020-07-22 21:12:17.695 debug1: no match: PuTTY_Release_0.73
5848 2020-07-22 21:12:17.695 debug1: Local version string SSH-2.0-OpenSSH_for_Windows_7.7
5848 2020-07-22 21:12:17.695 debug2: fd 3 setting O_NONBLOCK
5848 2020-07-22 21:12:17.976 debug3: spawning "C:\\Windows\\System32\\OpenSSH\\sshd.exe" "-y"
5848 2020-07-22 21:12:17.991 debug2: Network child is on pid 1504
5848 2020-07-22 21:12:17.991 debug3: send_rexec_state: entering fd = 6 config len 351
5848 2020-07-22 21:12:17.991 debug3: ssh_msg_send: type 0
5848 2020-07-22 21:12:17.991 debug3: send_rexec_state: done
5848 2020-07-22 21:12:17.991 debug3: ssh_msg_send: type 0
5848 2020-07-22 21:12:17.991 debug3: ssh_msg_send: type 0
5848 2020-07-22 21:12:17.991 debug3: preauth child monitor started
5848 2020-07-22 21:12:18.070 debug1: list_hostkey_types: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519 [preauth]
5848 2020-07-22 21:12:18.070 debug3: send packet: type 20 [preauth]
5848 2020-07-22 21:12:18.070 debug1: SSH2_MSG_KEXINIT sent [preauth]
5848 2020-07-22 21:12:18.070 debug3: receive packet: type 20 [preauth]
5848 2020-07-22 21:12:18.070 debug1: SSH2_MSG_KEXINIT received [preauth]
5848 2020-07-22 21:12:18.070 debug2: local server KEXINIT proposal [preauth]
5848 2020-07-22 21:12:18.070 debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1 [preauth]
5848 2020-07-22 21:12:18.070 debug2: host key algorithms: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519 [preauth]
5848 2020-07-22 21:12:18.070 debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
5848 2020-07-22 21:12:18.070 debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
5848 2020-07-22 21:12:18.070 debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
5848 2020-07-22 21:12:18.070 debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
5848 2020-07-22 21:12:18.070 debug2: compression ctos: none [preauth]
5848 2020-07-22 21:12:18.070 debug2: compression stoc: none [preauth]
5848 2020-07-22 21:12:18.070 debug2: languages ctos:  [preauth]
5848 2020-07-22 21:12:18.070 debug2: languages stoc:  [preauth]
5848 2020-07-22 21:12:18.070 debug2: first_kex_follows 0  [preauth]
5848 2020-07-22 21:12:18.070 debug2: reserved 0  [preauth]
5848 2020-07-22 21:12:18.070 debug2: peer client KEXINIT proposal [preauth]
5848 2020-07-22 21:12:18.070 debug2: KEX algorithms: curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,rsa2048-sha256,rsa1024-sha1,diffie-hellman-group1-sha1 [preauth]
5848 2020-07-22 21:12:18.070 debug2: host key algorithms: ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-rsa,ssh-dss [preauth]
5848 2020-07-22 21:12:18.070 debug2: ciphers ctos: aes256-ctr,aes256-cbc,rijndael-cbc@lysator.liu.se,aes192-ctr,aes192-cbc,aes128-ctr,aes128-cbc,chacha20-poly1305@openssh.com,3des-ctr,3des-cbc,blowfish-ctr,blowfish-cbc,arcfour256,arcfour128 [preauth]
5848 2020-07-22 21:12:18.070 debug2: ciphers stoc: aes256-ctr,aes256-cbc,rijndael-cbc@lysator.liu.se,aes192-ctr,aes192-cbc,aes128-ctr,aes128-cbc,chacha20-poly1305@openssh.com,3des-ctr,3des-cbc,blowfish-ctr,blowfish-cbc,arcfour256,arcfour128 [preauth]
5848 2020-07-22 21:12:18.070 debug2: MACs ctos: hmac-sha2-256,hmac-sha1,hmac-sha1-96,hmac-md5,hmac-sha2-256-etm@openssh.com,hmac-sha1-etm@openssh.com,hmac-sha1-96-etm@openssh.com,hmac-md5-etm@openssh.com [preauth]
5848 2020-07-22 21:12:18.070 debug2: MACs stoc: hmac-sha2-256,hmac-sha1,hmac-sha1-96,hmac-md5,hmac-sha2-256-etm@openssh.com,hmac-sha1-etm@openssh.com,hmac-sha1-96-etm@openssh.com,hmac-md5-etm@openssh.com [preauth]
5848 2020-07-22 21:12:18.070 debug2: compression ctos: none,zlib,zlib@openssh.com [preauth]
5848 2020-07-22 21:12:18.070 debug2: compression stoc: none,zlib,zlib@openssh.com [preauth]
5848 2020-07-22 21:12:18.070 debug2: languages ctos:  [preauth]
5848 2020-07-22 21:12:18.070 debug2: languages stoc:  [preauth]
5848 2020-07-22 21:12:18.070 debug2: first_kex_follows 0  [preauth]
5848 2020-07-22 21:12:18.070 debug2: reserved 0  [preauth]
5848 2020-07-22 21:12:18.070 debug1: kex: algorithm: curve25519-sha256@libssh.org [preauth]
5848 2020-07-22 21:12:18.070 debug1: kex: host key algorithm: ssh-ed25519 [preauth]
5848 2020-07-22 21:12:18.070 debug1: kex: client->server cipher: aes256-ctr MAC: hmac-sha2-256 compression: none [preauth]
5848 2020-07-22 21:12:18.070 debug1: kex: server->client cipher: aes256-ctr MAC: hmac-sha2-256 compression: none [preauth]
5848 2020-07-22 21:12:18.070 debug1: expecting SSH2_MSG_KEX_ECDH_INIT [preauth]
5848 2020-07-22 21:12:18.070 debug3: receive packet: type 30 [preauth]
5848 2020-07-22 21:12:18.070 debug3: mm_key_sign entering [preauth]
5848 2020-07-22 21:12:18.070 debug3: mm_request_send entering: type 6 [preauth]
5848 2020-07-22 21:12:18.070 debug3: mm_key_sign: waiting for MONITOR_ANS_SIGN [preauth]
5848 2020-07-22 21:12:18.070 debug3: mm_request_receive_expect entering: type 7 [preauth]
5848 2020-07-22 21:12:18.070 debug3: mm_request_receive entering [preauth]
5848 2020-07-22 21:12:18.070 debug3: mm_request_receive entering
5848 2020-07-22 21:12:18.070 debug3: monitor_read: checking request 6
5848 2020-07-22 21:12:18.070 debug3: mm_answer_sign
5848 2020-07-22 21:12:18.070 debug3: mm_answer_sign: hostkey proof signature ************(83)
5848 2020-07-22 21:12:18.070 debug3: mm_request_send entering: type 7
5848 2020-07-22 21:12:18.070 debug2: monitor_read: 6 used once, disabling now
5848 2020-07-22 21:12:18.070 debug3: send packet: type 31 [preauth]
5848 2020-07-22 21:12:18.070 debug3: send packet: type 21 [preauth]
5848 2020-07-22 21:12:18.070 debug2: set_newkeys: mode 1 [preauth]
5848 2020-07-22 21:12:18.070 debug1: rekey after 4294967296 blocks [preauth]
5848 2020-07-22 21:12:18.070 debug1: SSH2_MSG_NEWKEYS sent [preauth]
5848 2020-07-22 21:12:18.070 debug1: expecting SSH2_MSG_NEWKEYS [preauth]
5848 2020-07-22 21:12:18.085 debug3: receive packet: type 21 [preauth]
5848 2020-07-22 21:12:18.085 debug1: SSH2_MSG_NEWKEYS received [preauth]
5848 2020-07-22 21:12:18.085 debug2: set_newkeys: mode 0 [preauth]
5848 2020-07-22 21:12:18.085 debug1: rekey after 4294967296 blocks [preauth]
5848 2020-07-22 21:12:18.085 debug1: KEX done [preauth]
5848 2020-07-22 21:12:18.085 debug3: receive packet: type 5 [preauth]
5848 2020-07-22 21:12:18.085 debug3: send packet: type 6 [preauth]
5848 2020-07-22 21:13:34.085 debug3: receive packet: type 50 [preauth]
5848 2020-07-22 21:13:34.085 debug1: userauth-request for user mcertini service ssh-connection method none [preauth]
5848 2020-07-22 21:13:34.085 debug1: attempt 0 failures 0 [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_getpwnamallow entering [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_send entering: type 8 [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_getpwnamallow: waiting for MONITOR_ANS_PWNAM [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_receive_expect entering: type 9 [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_receive entering [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_receive entering
5848 2020-07-22 21:13:34.085 debug3: monitor_read: checking request 8
5848 2020-07-22 21:13:34.085 debug3: mm_answer_pwnamallow
5848 2020-07-22 21:13:34.085 debug2: parse_server_config: config reprocess config len 351
5848 2020-07-22 21:13:34.085 debug3: mm_answer_pwnamallow: sending MONITOR_ANS_PWNAM: 1
5848 2020-07-22 21:13:34.085 debug3: mm_request_send entering: type 9
5848 2020-07-22 21:13:34.085 debug2: monitor_read: 8 used once, disabling now
5848 2020-07-22 21:13:34.085 debug2: input_userauth_request: setting up authctxt for mcertini [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_inform_authserv entering [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_send entering: type 4 [preauth]
5848 2020-07-22 21:13:34.085 debug2: input_userauth_request: try method none [preauth]
5848 2020-07-22 21:13:34.085 debug3: userauth_finish: failure partial=0 next methods="publickey,keyboard-interactive" [preauth]
5848 2020-07-22 21:13:34.085 debug3: send packet: type 51 [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_receive entering
5848 2020-07-22 21:13:34.085 debug3: monitor_read: checking request 4
5848 2020-07-22 21:13:34.085 debug3: mm_answer_authserv: service=ssh-connection, style=
5848 2020-07-22 21:13:34.085 debug2: monitor_read: 4 used once, disabling now
5848 2020-07-22 21:13:34.085 debug3: receive packet: type 50 [preauth]
5848 2020-07-22 21:13:34.085 debug1: userauth-request for user mcertini service ssh-connection method publickey [preauth]
5848 2020-07-22 21:13:34.085 debug1: attempt 1 failures 0 [preauth]
5848 2020-07-22 21:13:34.085 debug2: input_userauth_request: try method publickey [preauth]
5848 2020-07-22 21:13:34.085 debug1: userauth_pubkey: test pkalg ssh-rsa pkblob RSA SHA256:******* [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_key_allowed entering [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_send entering: type 22 [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_key_allowed: waiting for MONITOR_ANS_KEYALLOWED [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_receive_expect entering: type 23 [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_receive entering [preauth]
5848 2020-07-22 21:13:34.085 debug3: mm_request_receive entering
5848 2020-07-22 21:13:34.085 debug3: monitor_read: checking request 22
5848 2020-07-22 21:13:34.085 debug3: mm_answer_keyallowed entering
5848 2020-07-22 21:13:34.085 debug3: mm_answer_keyallowed: key_from_blob: *********
5848 2020-07-22 21:13:34.085 debug1: trying public key file C:/users/mcertini/.ssh/authorized_keys
5848 2020-07-22 21:13:34.102 debug1: C:/users/mcertini/.ssh/authorized_keys:2: matching key found: RSA SHA256:******
5848 2020-07-22 21:13:34.102 debug1: C:/users/mcertini/.ssh/authorized_keys:2: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
5848 2020-07-22 21:13:34.102 Accepted key RSA SHA256:***** found at C:/users/mcertini/.ssh/authorized_keys:2
5848 2020-07-22 21:13:34.102 debug3: mm_answer_keyallowed: publickey authentication test: RSA key is allowed
5848 2020-07-22 21:13:34.102 debug3: mm_request_send entering: type 23
5848 2020-07-22 21:13:34.102 debug3: send packet: type 60 [preauth]
5848 2020-07-22 21:13:34.102 debug2: userauth_pubkey: authenticated 0 pkalg ssh-rsa [preauth]
5848 2020-07-22 21:13:34.102 Postponed publickey for mcertini from 192.168.1.170 port 51461 ssh2 [preauth]
5848 2020-07-22 21:13:45.978 debug3: receive packet: type 50 [preauth]
5848 2020-07-22 21:13:45.978 debug1: userauth-request for user mcertini service ssh-connection method publickey [preauth]
5848 2020-07-22 21:13:45.978 debug1: attempt 2 failures 0 [preauth]
5848 2020-07-22 21:13:45.978 debug2: input_userauth_request: try method publickey [preauth]
5848 2020-07-22 21:13:45.978 debug3: userauth_pubkey: have ssh-rsa signature for RSA SHA256:****** [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_key_allowed entering [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_request_send entering: type 22 [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_key_allowed: waiting for MONITOR_ANS_KEYALLOWED [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_request_receive_expect entering: type 23 [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_request_receive entering [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_request_receive entering
5848 2020-07-22 21:13:45.978 debug3: monitor_read: checking request 22
5848 2020-07-22 21:13:45.978 debug3: mm_answer_keyallowed entering
5848 2020-07-22 21:13:45.978 debug3: mm_answer_keyallowed: key_from_blob: ********
5848 2020-07-22 21:13:45.978 debug1: trying public key file C:/users/mcertini/.ssh/authorized_keys
5848 2020-07-22 21:13:45.978 debug1: C:/users/mcertini/.ssh/authorized_keys:2: matching key found: RSA SHA256:********
5848 2020-07-22 21:13:45.978 debug1: C:/users/mcertini/.ssh/authorized_keys:2: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
5848 2020-07-22 21:13:45.978 Accepted key RSA SHA256:***** found at C:/users/mcertini/.ssh/authorized_keys:2
5848 2020-07-22 21:13:45.978 debug3: mm_answer_keyallowed: publickey authentication: RSA key is allowed
5848 2020-07-22 21:13:45.978 debug3: mm_request_send entering: type 23
5848 2020-07-22 21:13:45.978 debug3: mm_sshkey_verify entering [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_request_send entering: type 24 [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_sshkey_verify: waiting for MONITOR_ANS_KEYVERIFY [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_request_receive_expect entering: type 25 [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_request_receive entering [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_request_receive entering
5848 2020-07-22 21:13:45.978 debug3: monitor_read: checking request 24
5848 2020-07-22 21:13:45.978 debug3: mm_answer_keyverify: publickey ******* signature verified
5848 2020-07-22 21:13:45.978 debug1: auth_activate_options: setting new authentication options
5848 2020-07-22 21:13:45.978 debug3: mm_request_send entering: type 25
5848 2020-07-22 21:13:45.978 Accepted publickey for mcertini from 192.168.1.170 port 51461 ssh2: RSA SHA256:******
5848 2020-07-22 21:13:45.978 debug1: monitor_child_preauth: mcertini has been authenticated by privileged process
5848 2020-07-22 21:13:45.978 debug3: mm_get_keystate: Waiting for new keys
5848 2020-07-22 21:13:45.978 debug3: mm_request_receive_expect entering: type 26
5848 2020-07-22 21:13:45.978 debug3: mm_request_receive entering
5848 2020-07-22 21:13:45.978 debug3: mm_get_keystate: GOT new keys
5848 2020-07-22 21:13:45.978 debug1: auth_activate_options: setting new authentication options [preauth]
5848 2020-07-22 21:13:45.978 debug2: userauth_pubkey: authenticated 1 pkalg ssh-rsa [preauth]
5848 2020-07-22 21:13:45.978 debug3: send packet: type 52 [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_request_send entering: type 26 [preauth]
5848 2020-07-22 21:13:45.978 debug3: mm_send_keystate: Finished sending state [preauth]
5848 2020-07-22 21:13:45.978 debug1: monitor_read_log: child log fd closed
5848 2020-07-22 21:13:45.978 debug1: generate_s4u_user_token: LsaLogonUser() failed. User 'mcertini' Status: 0xC000015B SubStatus 0.
5848 2020-07-22 21:13:45.978 debug3: get_user_token - unable to generate token for user mcertini
5848 2020-07-22 21:13:45.978 debug1: generate_s4u_user_token: LsaLogonUser() failed. User 'mcertini' Status: 0xC000015B SubStatus 0.
5848 2020-07-22 21:13:45.978 error: get_user_token - unable to generate token on 2nd attempt for user mcertini
5848 2020-07-22 21:13:45.978 error: unable to get security token for user mcertini
5848 2020-07-22 21:13:45.978 fatal: fork of unprivileged child failed
5848 2020-07-22 21:13:45.978 debug1: do_cleanup

**Comment by jborean93** [comment]
The error `0xC000015B` means `STATUS_LOGON_TYPE_NOT_GRANTED` which means that the user you are trying to log on as has not been granted the logon type that is requested. In this case for an S4U logon (which is what SSH key auth uses) will request a `Network` logon type as per [this line](https://github.com/PowerShell/openssh-portable/blob/8ab565c53f3619d6a1f5ac229e212cad8a52852c/contrib/win32/win32compat/win32_usertoken_utils.c#L202-L204).

This would mean that the user either has [SeDenyNetworkLogonRight](https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/deny-access-to-this-computer-from-the-network) which stops that account from being logged on as a network logon. Or it does not have the [SeNetworkLogonRight](https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/access-this-computer-from-the-network) which allows access from a network. A deny rule will take preference if the user has both rights.

You can view these rights and what users/groups have them by running `gpedit.msc` and going to `Local Computer Policy` -> `Computer Configuration` -> `Software Settings` -> `Security Settings` -> `Local Policies` -> `User Rights Assignment`.

**Comment by mcertini** [comment]
jborean93
Thank you for the insight, I have been working on this for many days.  Finally found clarity.  The path though you specified above did not work for me.  I did not have anything listed under Software Settings.  Where I found the settings though were:
Local Security Policy -> Security Settings -> Local Policies -> User Rights Assignment -> .  Here though I did not know what settings to change.  I had to do a search with the error message to find the website below that described the proper policy setting to use:
https://www.bitvise.com/ssh-server-guide-logon-type  (Look for Windows Security Settings Affecting Logon Types)
